### PR TITLE
Updated to support gpt-4o

### PR DESCRIPTION
**Pull request by toshiakit**
I updated README
openAIChat
and one example that contained a bug. 
All examples were tested and ran without errors.

### CODE REVIEW

**Review by vpapanasta — CHANGES_REQUESTED** [review]
It would be great to add a unit test for the new model in topenAIChat using a secret key. See as example the test point "createOpenAIChatWithOpenAIKey"

**Comment by toshiakit** [comment]
> It would be great to add a unit test for the new model in topenAIChat using a secret key. See as example the test point "createOpenAIChatWithOpenAIKey"

Done

**Comment by toshiakit** [comment]
> I have not reviewed the mlx file change.

There was an error when I tested the example. The variable name was renamed to "documents" except in one place and that caused the error.

File: openAIChat.m [hunk 1]
```diff
@@ -217,7 +218,7 @@
             end
 
             if iscell(messagesStruct{end}.content) && any(cellfun(@(x) isfield(x,"image_url"), messagesStruct{end}.content))
-                if ~ismember(this.ModelName,["gpt-4-turbo","gpt-4-turbo-2024-04-09"]) 
+                if ~ismember(this.ModelName,["gpt-4-turbo","gpt-4-turbo-2024-04-09","gpt-4o","gpt-4o-2024-05-13"])
```
[the remark this answers is not in the document]
Comment (toshiakit) [reply openAIChat.m:221]: Please create a separate pull request to address this. This pull request is meant to bring this repo current with the latest OpenAI release.

**Review by ccreutzi — APPROVED** [review]
Thanks!

File: tests/topenAIChat.m [hunk 2]
```diff
@@ -159,6 +159,11 @@ function createOpenAIChatWithOpenAIKey(testCase)
                 ApiKey=getenv("OPENAI_KEY"));
 
             testCase.verifyWarningFree(@()generate(chat, "Hello world."));
+
+            chat = openAIChat("You are a useful assistant", ...
+                ApiKey=getenv("OPENAI_KEY"), ModelName="gpt-4o");
+
+            testCase.verifyWarningFree(@()generate(chat, "Hello world."));
```
Comment (vpapanasta) [inline tests/topenAIChat.m:166]: It is best to either create a new test point for "gpt-4o" or create a test parameter and with the default model value and the new "gpt-4o" one.

**Review by vpapanasta — COMMENTED** [review]
Thanks, Toshi, I left some extra feedback.

File: tests/topenAIChat.m (hunk 2 above)
Comment (toshiakit) [reply tests/topenAIChat.m]: I created a separate test point.

**Review by vpapanasta — APPROVED** [review]
Thanks for the changes Toshi.

**Pull request merged**